lib/matplotlib/collections.py

```diff
@@ -1242,7 +1242,7 @@ def set_verts(self, verts, closed=True):
             return
 
         # Fast path for arrays
-        if isinstance(verts, np.ndarray) and len(verts.shape) == 3:
+        if isinstance(verts, np.ndarray) and len(verts.shape) == 3 and len(verts) > 0:
```

Comment:
```suggestion
        if isinstance(verts, np.ndarray) and len(verts.shape) == 3 and verts.size:
```
?

(I guess that len may mess up in some special cases...)